nethack/src/board.c

```diff
@@ -0,0 +1,226 @@
+/* vim:set cin ft=c sw=4 sts=4 ts=8 et ai cino=Ls\:0t0(0 : -*- mode:c;fill-column:80;tab-width:8;c-basic-offset:4;indent-tabs-mode:nil;c-file-style:"k&r" -*-*/
+/* NetHack may be freely redistributed.  See license for details. */
+
+#include "nhcurses.h"
+#include <time.h>
+
+struct gamewin *gw;
+struct win_menu *mdat;
+nh_bool done;
+struct nh_board_entry *entries;
+int listlen, maxplayersize;
+
+static void
+board_display_curses_ui(struct nh_menuitem *items, int icount)
+{
+    int startx, starty, x1, x2, y1, y2;
+
+    if (gw != NULL) {
+        delwin(mdat->content);
+        delwin(gw->win);
+        delete_gamewin(gw);
+    }
+
+    gw = alloc_gamewin(sizeof (struct win_menu));
+    gw->draw = draw_menu;
+    gw->resize = resize_menu;
+    mdat = (struct win_menu *)gw->extra;
+    mdat->items = items;
+    mdat->icount = icount;
+    mdat->title = "Binge Board";
+    mdat->selected = calloc(icount, sizeof (nh_bool));
```

Comment:
You used xmalloc in a previous file, which checks for errors. Either use that here too, or check for errors.